inference-engine/thirdparty/clDNN/api/cldnn/primitives/implementation_desc.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "cldnn/runtime/tensor.hpp"
+
+#include <map>
+#include <ostream>
+
+namespace cldnn {
+
+/// @brief Primitives implementation type.
+enum class impl_types : uint8_t {
+    cpu = 1 << 0,
+    common = 1 << 1,
+    ocl = 1 << 2,
+    any = 0xFF,
```

Comment:
Is my understanding correct that "any" is actually "nothing is registered " though it sounds like "not preferring specific impl" ?

Reply:
Self answer) Above was not true, and seems that "any" is "not preferring specific impl" and it will select impl with the order of the registered map :)